uber/models.py

```diff
@@ -1076,9 +1070,9 @@ def _badge_adjustments(self):
         # _assert_badge_lock()
 
         if self.badge_type in [c.PSEUDO_GROUP_BADGE, c.PSEUDO_DEALER_BADGE]:
```

Comment:
Another bug found by the unit tests.  We were changing their badge before assigning them the ribbon, at which point ``is_dealer`` would no longer be true.